### PR TITLE
feat: add RTCM3 packet extraction functionality

### PR DESCRIPTION
**Pull request by doudou**
It's just enough to extract RTCM3 packets from another stream, to e.g.
save it or forward it to another GNSS unit.

### CODE REVIEW

File: test/test_rtcm3.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    const std::vector<uint8_t> buffer =
+    {
+        0xd3, 0x00, 0x13, 0x3e, 0xd0, 0x00, 0x02, 0x36,
+        0xfd, 0xb8, 0x0d, 0xde, 0x08, 0x00, 0x5b, 0x2b,
+        0xc1, 0x08, 0xa7, 0xb9, 0x8d, 0x3d, 0xd8, 0xab, 0x37
+    };
+    BOOST_TEST(rtcm3::extractPacket(buffer.data(), 25) == 25);
+}
+
+//BOOST_AUTO_TEST_CASE(it_should_parse_RTCM3) {
```
Comment (g-arjones) [inline test/test_rtcm3.cpp:84]: Why the commented out test?

File: src/rtcm3.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+     */
+    namespace rtcm3 {
+        static const std::uint8_t PREAMBLE = 0xD3;
+        static const int PREAMBLE_SIZE = 1;
+        static const int HEADER_SIZE = 3;
+        static const int CRC_SIZE = 3;
+        static const int MIN_PACKET_SIZE = HEADER_SIZE + CRC_SIZE;
+
+        bool isPreamble(std::uint8_t const* buffer, std::size_t size);
+        uint16_t getLength(uint8_t const* buffer, std::size_t size);
```
Comment (g-arjones) [inline src/rtcm3.hpp:19]: Would you mind changing to `std::uint8_t` for the sake of consistency?

**Review by g-arjones — APPROVED** [review]
I would have added APIs that allow passing a `std::vector<uint8_t>` as well but otherwise looks good to me.

File: test/test_rtcm3.cpp (hunk 1 above)
Comment (doudou) [reply test/test_rtcm3.cpp:84]: Was a leftover. I started by running the code on a RTCM log I found in rtklib. Will remove

File: src/rtcm3.hpp (hunk 2 above)
Comment (doudou) [reply src/rtcm3.hpp]: On the contrary. Done

**Review by g-arjones — APPROVED** [review]

**Pull request merged**